### PR TITLE
SONAR-8746 Remove measures related to periods 2 to 5

### PR DESCRIPTION
**Pull request by julienlancelot**

### CODE REVIEW

**Comment by simonbrandhof** [comment]
The PR should be based on branch-6.3, not master.

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v63/DeleteMeasuresHavingNoValue.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+  protected void execute(Context context) throws SQLException {
+    MassUpdate massUpdate = context.prepareMassUpdate();
+    massUpdate.select("SELECT pm.id FROM project_measures pm " +
+      "WHERE pm.value IS NULL AND pm.text_value IS NULL AND pm.variation_value_1 IS NULL AND pm.measure_data IS NULL ");
+    massUpdate.update("DELETE FROM project_measures WHERE id=?");
+    massUpdate.rowPluralName("measures");
+    massUpdate.execute(this::handle);
+  }
+
+  private boolean handle(Select.Row row, SqlStatement update) throws SQLException {
```
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v63/DeleteMeasuresHavingNoValue.java]: MAJOR Make "handle" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Comment by SonarTech** [comment]
SonarQube analysis reported 1 issue
* ![MAJOR][MAJOR] 1 major

Watch the comments in this conversation to review them.

[MAJOR]: https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR'

File: server/sonar-db-migration/src/test/java/org/sonar/server/platform/db/migration/version/v63/DeleteMeasuresHavingNoValueTest.java [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    db.commit();
+
+    underTest.execute();
+    assertThat(db.countRowsOfTable(TABLE_MEASURES)).isEqualTo(2);
+
+    underTest.execute();
+    assertThat(db.countRowsOfTable(TABLE_MEASURES)).isEqualTo(2);
+  }
+
+  private void insertMeasure(@Nullable Double value, @Nullable String textValue, @Nullable Double variation, @Nullable String data) {
```
Comment (julienlancelot) [inline server/sonar-db-migration/src/test/java/org/sonar/server/platform/db/migration/version/v63/DeleteMeasuresHavingNoValueTest.java]: Ok, I'll add some tests

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v63/DeleteMeasuresHavingNoValue.java [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+import java.sql.SQLException;
+import org.sonar.db.Database;
+import org.sonar.server.platform.db.migration.step.DataChange;
+import org.sonar.server.platform.db.migration.step.MassUpdate;
+import org.sonar.server.platform.db.migration.step.Select;
+import org.sonar.server.platform.db.migration.step.SqlStatement;
+
+/**
+ * Delete measures having no value in order to remove measures related to periods 2 to 5
```
Comment (julienlancelot) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v63/DeleteMeasuresHavingNoValue.java]: ok

File: server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v63/DeleteMeasuresHavingNoValue.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+    massUpdate.select("SELECT pm.id FROM project_measures pm " +
+      "WHERE pm.value IS NULL AND pm.text_value IS NULL AND pm.variation_value_1 IS NULL AND pm.measure_data IS NULL ");
+    massUpdate.update("DELETE FROM project_measures WHERE id=?");
+    massUpdate.rowPluralName("measures");
+    massUpdate.execute((row, update) -> {
+      update.setLong(1, row.getLong(1));
+      return true;
+    });
+  }
+
```
Comment (SonarTech) [inline server/sonar-db-migration/src/main/java/org/sonar/server/platform/db/migration/version/v63/DeleteMeasuresHavingNoValue.java]: MAJOR Make "handle" a "static" method. [rule](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325)

**Pull request merged**